Utility.cpp

```diff
@@ -1,108 +1,141 @@
 #include "Utility.h"
-
 #include "HelpfulItem.h"
 #include "DefensiveItem.h"
 #include "Character.h"
+#include "Dwarf.h"
+#include "Dragon.h"
+#include "DragonSlayer.h"
+#include "Paladin.h"
+
+#include <random>
 
+const int MIN_RANDOM_ITEM = 1;
+const int MAX_RANDOM_ITEM = 5;
 
 std::vector<std::unique_ptr<Item>> makeHelpfulItems(int num)
 {
     std::vector<std::unique_ptr<Item>> items;
-    
-    while( num-- >= 0 )
+
+    while (num-- > 0)
     {
-        items.push_back( std::unique_ptr<HelpfulItem>(new HelpfulItem()) );
+        items.push_back(std::unique_ptr<HelpfulItem>(new HelpfulItem()));
     }
-    
+
     std::cout << "made " << items.size() << " helpful items" << std::endl;
     return items;
 }
 
 std::vector<std::unique_ptr<Item>> makeDefensiveItems(int num)
 {
     std::vector<std::unique_ptr<Item>> items;
-    
-    while( num-- >= 0 )
+
+    while (num-- > 0)
     {
-        items.push_back( std::unique_ptr<DefensiveItem>(new DefensiveItem()) );
+        items.push_back(std::unique_ptr<DefensiveItem>(new DefensiveItem()));
     }
-    
+
     std::cout << "made " << items.size() << " defensive items" << std::endl;
     return items;
 }
 
+std::vector<std::unique_ptr<Item>> makeRandomHelpfulItems()
+{
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_int_distribution<int> dist(MIN_RANDOM_ITEM, MAX_RANDOM_ITEM);
+
+    return makeHelpfulItems(dist(gen));
+}
+
+std::vector<std::unique_ptr<Item>> makeRandomDefensiveItems()
+{
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_int_distribution<int> dist(MIN_RANDOM_ITEM, MAX_RANDOM_ITEM);
+
+    return makeDefensiveItems(dist(gen));
+}
```

Comment:
This seems overly complicated.  Why not simply call `make_____Items(rand())` in the derived Character constructors?